sycl/doc/design/CommandGraph.md

```diff
@@ -132,6 +132,20 @@ in a command-graph, it will perform a blocking wait on the dependencies of the
 command-group first. The user will experience this wait as part of graph
 finalization.
 
+## Graph partitioning
+
+To handle dependencies from other devices, the graph can be partionned during the finalization process. A partition is a set of one or more nodes intended to run on the same device. Each partition instanciate a CommandBuffer (or equivalent) which contains all the commands to be executed on the device. Therefore, the partitioning only impacts `exec_graph`. Synchronizations between partitions are managed by the runtime unlike internal partition dependencies that are handled direclyt by the backend.
+Since runtime synchronization and multiple CommandBuffer involves extra latency, the implementation ensures to minimize the number of partitions.
+Currently, the creation of a new partition is triggered by a node containing an `host_task`.
+When a host_task is encountered the predecessors of this `host_task` node are assigned to one partition, the `host_task` is assigned to another partition, and the successors are assigned to a third partition as shown below:
+
+![Graph partition illustration.](images/SYCL-Graph-partitions.jpg)
+
+Partition numbers are allocated in order. Hence, the runtime must ensure that Partition `n` complete before strating execution of Partition `n+1`.
+
+Note that partitioning can only happen during the finalization stage due to potential backward dependencies that could be created using the `make_edge` function.
+
+
```

Comment:
Leaving a comment here because it's documentation related, but the spec needs updated to say that host-tasks are supported https://github.com/reble/llvm/blob/6d5a347495954dc1826ca10e4210f42ebf37ebe6/sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc#710-host-tasks